File: subarraysumeqsk3.cpp
class Solution
{
public:
    int subarraySum(vector<int> &nums, int k)
    {
        map<int, int> smap;
        smap[0] = 1;
        int sum = 0;
        int count = 0;

        for (int num : nums)
        {
            sum += num;
            if (smap.find(sum - k) != smap.end())
            {
                count += smap[sum - k];
            }
            smap[sum]++;
        }

        return count;
    }
};

File: subarraysumeqsk3.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> smap = new HashMap<>();

        int count = 0;
        int sum = 0;
        smap.put(0, 1);

        for (int num : nums) {
            sum += num;

            if (smap.containsKey(sum - k)) {
                count += smap.get(sum - k);
            }

            smap.put(sum, smap.getOrDefault(sum, 0) + 1);
        }

        return count;
    }
}